src/components/product/Products.js

```diff
@@ -19,7 +22,7 @@ const Products = () => {
 
   const getProductUpdate = (id) => {
     const product = state.product.filter((prod) => prod.id === id);
-    setProduct({ ...product });
```

Comment:
`filter` would be sent back an array
use `find` instead so that you just need to do something like setProduct(product ) because it returns an object of product.